### PR TITLE
Refactored tests

### PR DESCRIPTION
**Pull request by janoszen**
This pull request makes the `TestAccOvirtVM_basic` test runnable again by implementing methods that allow it to run on any oVirt cluster that has a working host.

The test suite supports the following environment variables (to be added to the README):

- `OVIRT_USERNAME`
- `OVIRT_PASSWORD`
- `OVIRT_URL`  (Should point to `https://api-host/ovirt-engine/api`)
- `OVIRT_INSECURE` (Set to 1 to connect the oVirt engine without certificate verification)
- `OVIRT_CAFILE` (Point to CA certificate file.)
- `OVIRT_CA_BUNDLE` (Provide CA certificate directly)
- `OVIRT_TEST_CLUSTER_ID` (Provide the cluster ID to run on. Defaults to the first cluster with a working host.)
- `OVIRT_BLANK_TEMPLATE_ID` (Provide the ID of the blank template. Defaults to the template with the description "Blank template")
- `OVIRT_TEST_STORAGEDOMAIN_ID` (Provide the ID of the storage domain to use for testing. Defaults to the first storage domain with at least 2 GB of space.)
- `OVIRT_TEST_IMAGE_PATH` (Provide a full path to an image to upload for testing. Defaults to `$CWD/ovirt/testimage/image`. You may need to adjust this if your current working directory is different.)

About the test image: since the upload can be a bottleneck and we don't want to rely on a bootable image existing the image located in the `ovirt/testimage` folder is a tiny  512 byte ASM program that prints `Hello oVirt!` on boot and then stops the CPU, but keeps the VM running so the Terraform provider can correctly identify it as running. (This PR also fixes image upload issues on Windows and when the image is smaller than 1 GB.) Further reading: https://www.briansteffens.com/2020/01/15/hello-world-from-a-bootloader.html

**TODO:** decide how many of the other tests we want to keep (Feel free to discuss below.)

### CODE REVIEW

File: ovirt/suite_test.go [hunk 1]
```diff
@@ -0,0 +1,773 @@
...
+			return err
+		}
+		if len(listResponse.MustClusters().Slice()) != 0 {
+			return fmt.Errorf("cluster %s has not been removed after test", cluster.MustName())
+		}
+		return nil
+	}
+}
+
+func (o *ovirtTestSuite) TestResourceAttrNotEqual(
```
Comment (eslutsky) [inline ovirt/suite_test.go:606]: looks like a very general-purpose method, do we want it inside the suite?

Comment (janoszen) [reply ovirt/suite_test.go:606]: It has been moved from an old method that was there. I generally moved everything into the suite to keep things in one place.

Comment (eslutsky) [reply ovirt/suite_test.go:606]: okay .. those kinds of methods are usually handled by testing frameworks, are we planning to integrate one here?

Comment (janoszen) [reply ovirt/suite_test.go:606]: No. The suite is the testing framework.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+Formatting changes, adding or removing empty lines should be avoided under all circumstances because they unnecessarily bloat the patch size. If you want to submit a formatting change please do so in a separate PR.
+
+### Use and write tests
+
+Tests are an integral part of writing a Terraform provider as manual testing every change is nearly impossible.
+
+You can run the oVirt provider tests using the following command:
+
+```bash
+OVIRT_URL=https://your-ovirt-engine-host/ovirt-engine/api/;OVIRT_USERNAME=admin@internal;OVIRT_PASSWORD=your-ovirt-password;TF_ACC=1;OVIRT_INSECURE=1 go test -v ./...
```
Comment (eslutsky) [inline CONTRIBUTING.md:28]: i couldnt run it with this command , ended up using : `IRT_URL=https://your-ovirt-engine-host/ovirt-engine/api/ OVIRT_USERNAME=admin@internal OVIRT_PASSWORD=your-ovirt-password TF_ACC=1 OVIRT_INSECURE=1 go test -v ./...`

**Comment by eslutsky** [comment]
these tests are failing for me

**Comment by eslutsky** [comment]
> @eslutsky what storage domain are you running on?

I'm using NFS , when specifying OVIRT_TEST_STORAGEDOMAIN_ID=<doman-id>   it passes

File: ovirt/data_source_ovirt_nics_test.go [hunk 3]
```diff
@@ -1,33 +1,98 @@
...
+  source_url = "%s"
+  storage_domain_id = "%s"
+  sparse = true
+}
+
+resource "ovirt_vm" "vm" {
+  name        = "%s"
+  cluster_id  = "%s"
+  template_id = "%s"
+  autostart   = false
```
Comment (eslutsky) [inline ovirt/data_source_ovirt_nics_test.go:45]: failing with:
```=== RUN   TestAccOvirtNicsDataSource_nameRegexFilter
    testing.go:654: Step 0 error: config is invalid: Unsupported argument: An argument named "autostart" is not expected here. Did you mean "auto_start"?
```    
please change this to "auto_start"

File: ovirt/resource_ovirt_cluster_test.go [hunk 4]
```diff
@@ -18,17 +18,18 @@ import (
 func TestAccOvirtCluster_basic(t *testing.T) {
 	datacenterID := "5bc08e5b-03ab-0194-03cb-000000000289"
```
Comment (eslutsky) [inline ovirt/resource_ovirt_cluster_test.go:19]: hardcoded datacenterID , failing with:
```**=== RUN   TestAccOvirtCluster_basic
    testing.go:654: Step 0 error: errors during apply:
        
        Error: Fault reason is "Operation Failed". Fault detail is "Entity not found: Datacenter: id=5bc08e5b-03ab-0194-03cb-000000000289". HTTP response code is "404". HTTP response message is "404 Not Found".
        
          on /tmp/tf-test131613786/main.tf line 2:
          (source code not available)
**```

File: ovirt/suite_test.go [hunk 5]
```diff
@@ -0,0 +1,773 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	if testImageSourcePath == "" {
+		cwd, err := os.Getwd()
+		if err != nil {
+			return nil, fmt.Errorf("can't fetch current working directory (%w)", err)
+		}
+		testImageSourcePath = path.Join(cwd, "ovirt", "testimage", "image")
```
Comment (eslutsky) [inline ovirt/suite_test.go:122]: === RUN   TestAccOvirtNicsDataSource_nameRegexFilter
    testing.go:654: Step 0 error: errors during apply:
        
        Error: Failed preparing disk for transfer: open /home/eslutsky/GolandProjects/terraform-provider-ovirt/ovirt/ovirt/testimage/image: no such file or directory
        
          on /tmp/tf-test360349721/main.tf line 2:
          (source code not available)
      
it works without  "ovirt" in  path:
`testImageSourcePath = path.Join(cwd, "testimage", "image")`

Comment (janoszen) [reply ovirt/suite_test.go:122]: Which folder did you run this from?

Comment (eslutsky) [reply ovirt/suite_test.go:122]: project root `/home/eslutsky/GolandProjects/terraform-provider-ovirt`

Comment (janoszen) [reply ovirt/suite_test.go:122]: Thanks, I'll look into that, it should work from the project root.

**Comment by janoszen** [comment]
@emesika all yours.

**Comment by janoszen** [comment]
@emesika can we get this PR going? I'm switching to a new GitHub account and I'd like to sort out this PR before that.

**Comment by janoszen** [comment]
@emesika @Gal-Zaidman @eslutsky I'm closing and re-opening this PR because I'm switching to a new GitHub account. The new PR will be up shortly, please proceed on that one.

**Pull request closed without merging**